File: 0x06-python-classes/6-square.py
#!/usr/bin/python3

"""Class Square with methods for size, area and printing"""


class Square:
    """Class Square with methods for size, area and printing"""

    @property
    def size(self):
        return self.__size

    @size.setter
    def size(self, value):
        if type(value) != int:
            raise TypeError("size must be an integer")
        if value < 0:
            raise ValueError("size must be >= 0")
        self.__size = value

    @property
    def position(self):
        return self.__position

    @position.setter
    def position(self, value):
        if type(value) != tuple:
            raise TypeError("position must be a tuple of 2 positive integers")
        if value[0] < 0 or value[1] < 0:
            raise TypeError("position must be a tuple of 2 positive integers")
        self.__position = value

    def __init__(self, size=0, position=(0, 0)):
        self.__size = size
        self.__position = position

    def area(self):
        return (self.__size * self.__size)

    def my_print(self):
        for i in range(0, self.__size):
#            if self.__position[1] > 0:
            for u in range(self.__position[0]):
                print(" ", end="")
            for j in range(0, self.__size):
                print("#", end="")
            print()

File: 0x13-javascript_objects_scopes_closures/6-square.js
#!/usr/bin/node

const BaseSquare = require('./5-square.js');

class Square extends BaseSquare {
  charPrint (c) {
    let p = '';

    for (let i = 0; i < this.size; i++) {
      for (let j = 0; j < this.size; j++) {
        if (!c) {
          p += 'X';
        } else {
          p += c;
        }
      }
      if (i + 1 < this.size) {
        p += '\n';
      }
    }
    console.log(p);
  }
}

module.exports = Square;
